discovery-infra/install_cluster.py

```diff
@@ -227,29 +234,58 @@ def download_logs_from_all_hosts(client, cluster_id, output_folder):
         )
 
 
+def get_nodes_number_from_cluster_deployment(cluster_deployment):
+    try:
+        spec = cluster_deployment.get()['spec']
+        provision = spec['installStrategy']['agent']['provisionRequirements']
+        return provision['controlPlaneAgents'] + provision['workerAgents']
+
+    except KeyError as key:
+        raise RuntimeError(
+            f'unable to get nodes number of cluster deployment: '
+            f'{cluster_deployment.ref}, missing key: {key}'
+        )
+
+
 def main():
     _verify_kube_download_folder(args.kubeconfig_path)
     log.info("Creating assisted service client")
     # if not cluster id is given, reads it from latest run
     tf = None
+    cluster_name = f'{args.cluster_name or consts.CLUSTER_PREFIX}-{args.namespace}'
     if not args.cluster_id:
-        cluster_name = f'{args.cluster_name or consts.CLUSTER_PREFIX}-{args.namespace}'
         tf_folder = utils.get_tf_folder(cluster_name, args.namespace)
         args.cluster_id = utils.get_tfvars(tf_folder).get('cluster_inventory_id')
         tf = terraform_utils.TerraformUtils(working_dir=tf_folder)
 
-    client = assisted_service_api.create_client(
-        url=utils.get_assisted_service_url_by_args(
-            args=args,
-            wait=False
+    client, cluster_deployment, nodes_number = None, None, None
+    if args.kube_api:
```

Comment:
can we move all kube_api flow to the new file? I don't see why we don't have kube_api.py with all the functions relevant to it?

Reply:
moved installation function to kubeapi_utils.py